color/src/commonMain/kotlin/Color.kt

```diff
@@ -0,0 +1,59 @@
+package com.juul.krayon.color
+
+import kotlin.math.roundToInt
+
+/** A color in ARGB color space. */
+public inline class Color(public val argb: Int) {
+
+    /** Create a color from component integers. Components are masked to their last eight bits. */
+    public constructor(alpha: Int, red: Int, green: Int, blue: Int) :
+        this(
+            ((alpha and MASK_COMPONENT) shl SHIFT_ALPHA) or
+                ((red and MASK_COMPONENT) shl SHIFT_RED) or
+                ((green and MASK_COMPONENT) shl SHIFT_GREEN) or
+                ((blue and MASK_COMPONENT) shl SHIFT_BLUE)
+        )
+
+    /** Create an opaque color from component integers. Components are masked to their last eight bits. */
+    public constructor(red: Int, green: Int, blue: Int) :
+        this(alpha = COMPONENT_MAX, red, green, blue)
+
+    /** Create a color from component floats. Components are multiplied by 255 and rounded. */
+    public constructor(alpha: Float, red: Float, green: Float, blue: Float) : this(
+        alpha.toColorComponent(), red.toColorComponent(), green.toColorComponent(), blue.toColorComponent()
+    )
+
+    /** Create an opaque color from component floats. Components are multiplied by 255 and rounded. */
+    public constructor(red: Float, green: Float, blue: Float) :
+        this(alpha = FLOAT_COMPONENT_MAX, red, green, blue)
```

Comment:
Since CSS (what is more familiar to JavaScript consumers) is `RGBA`, should we also provide a constructor that takes a string that supports web colors (e.g. `red`, `#f00`, `#FF0000FF`)?

Perhaps a helper function would help distinguish its purpose?

```kotlin
fun cssColor(color: String): Color = TODO()
```

Reply:
Perhaps we introduce CSS-style color support in a separate PR?

Reply:
I think a second PR is appropriate. Only reason I did the other modifications now (built in colors, lerp) was because I figured I might as well keep all breaking changes in one PR.

When the time comes though, how do you feel about simply `fun String.toColor()`?

Reply:
`String.toColor` sounds great.

Reply:
Created #22